src/jit/lsra.cpp

```diff
@@ -9884,9 +9884,18 @@ void LinearScan::resolveEdge(BasicBlock*      fromBlock,
         (resolveType == ResolveSharedCritical) ? REG_NA : getTempRegForResolution(fromBlock, toBlock, TYP_INT);
 #endif // !_TARGET_XARCH_
     regNumber tempRegFlt = REG_NA;
+#ifdef _TARGET_ARM_
+    // ARM32 requires double temp register in addition to interger and float temp registers.
+    regNumber tempRegDbl = REG_NA;
+#endif
     if ((compiler->compFloatingPointUsed) && (resolveType != ResolveSharedCritical))
     {
         tempRegFlt = getTempRegForResolution(fromBlock, toBlock, TYP_FLOAT);
+
+#ifdef _TARGET_ARM_
+        // ARM32 requires double temp register in addition to interger and float temp registers.
+        tempRegDbl = getTempRegForResolution(fromBlock, toBlock, TYP_DOUBLE);
```

Comment:
If you checked for a `TYP_DOUBLE` first, and it was found, then you could save its lower half as the `TYP_FLOAT` temp. If not, then you could check for a single register for `TYP_FLOAT`